lib/src/utils/product.dart

```diff
@@ -1,17 +1,40 @@
+import 'dart:convert';
+import 'package:flutter/foundation.dart';
 
 class Product {
-  String id;
-  String price;
-  String title;
-  String currencyCode;
+  late String id;
+  late String title;
+  late String currencyCode;
+  late double price;
+  late String priceString;
+  late SubscriptionPeriod subscriptionPeriod;
 
-  Product(this.id, this.price, this.title, this.currencyCode);
+  Product(this.id, this.price, this.priceString, this.title, this.currencyCode, this.subscriptionPeriod);
 
   factory Product.fromJson(dynamic json) {
-    print(json);
+    if(kDebugMode) print(json);
+    var subscriptionPeriod = new SubscriptionPeriod.fromMap(json['subscriptionPeriod'] as Map<String, dynamic>);
+    return Product(json['productId'] as String, json['productPrice'] as double, json['productPriceString'] as String,
+        json['productTitle'] as String, json['currencyCode'] as String, subscriptionPeriod);
+  }
+
+  @override
+  String toString() {
+      return 'Product(id: $id, price: $price, priceString: $priceString title: $title, currencyCode: $currencyCode, subscriptionPeriod: $subscriptionPeriod)';
+  }
+}
 
-    return Product(json['productId'] as String, json['productPrice'] as String,
-        json['productTitle'] as String, json['currencyCode'] as String);
+class SubscriptionPeriod {
+  /// unit represent the duration of an interval, from a day up to a year.
+  /// For example, unit value would be a month, year, day and week.
+  late String unit;
+  /// The number of units per subscription period.
+  /// For example, if the number of units is 6, then the subscription period would be 6 months.
+  late int numberOfUnits;
+
+  SubscriptionPeriod.fromMap(Map<String, dynamic> map) {
+    unit = map['periodUnit'].toString();
+    numberOfUnits = map['numberOfUnits'] as int;
```

Comment:
In iOS, we are returning `nil` if the number of units is nil. How would it affect this?